src/standaloneModal.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useRef, useState } from 'react'
+import { type ModalComponent, type ModalName, manager } from './manager'
+
+
+// to render a modal in-place without registration required
+const standaloneModal = (modalName: ModalName, Component: ModalComponent) => {
+  // ATTN because we can render several instances of standaloneModal,
+  //  we should track them and render only in the active one
+  let lastId = 1
+  let activeIds: number[] = []
+
+  const Wrapper: React.FC = () => {
+    const [ , forceUpdate ] = useState(0)
+    const instanceIdRef = useRef<number>(0)
+
+    if (!instanceIdRef.current) {
+      instanceIdRef.current = lastId++
+    }
+
+    useEffect(() => {
+      const listener = () => {
+        forceUpdate((value) => (value + 1))
+      }
+      manager.subscribe(listener)
+      manager.markAsStandalone(modalName)
+
+      activeIds.push(instanceIdRef.current)
+
+      return () => {
+        activeIds = activeIds.filter((value) => value !== instanceIdRef.current)
+        manager.unsubscribe(listener)
+        manager.unmarkAsStandalone(modalName)
+      }
+    }, [])
+
+    // ATTN show only if this instance is the first instance of this modal
+    if (activeIds[0] !== instanceIdRef.current) {
+      return null
+    }
+
+    let nodes = []
+
+    for (const [ id, { name, props, closeModal } ] of manager.getState().entries()) {
+      if (modalName === name) {
+        nodes.push(
+          <Component
+            key={id}
+            name={name}
+            closeModal={closeModal}
+            {...props}
+          />
+        )
+      }
+    }
```

Comment:
why don't we need render only isStandalone modals?

Reply:
All the system still built on unique modal names.
standaloneModal keep a modal Component and it's name in closure, it knows that it's standalone by name and verifies the name of opened modal in the state.
ModalRenderer knows nothing about modals and renders all the state (that keep all modals, global and standalone) by picking Component from the registry, that's why ModalRenderer should check `isStandalone` and skip them.